plugins/resteasy-http/src/main/java/org/radargun/service/RESTEasyHTTPOperations.java

```diff
@@ -0,0 +1,263 @@
+package org.radargun.service;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.UnsupportedEncodingException;
+import java.net.InetSocketAddress;
+import java.net.URLEncoder;
+
+import javax.ws.rs.HttpMethod;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation;
+import javax.ws.rs.core.EntityTag;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+import org.radargun.logging.Log;
+import org.radargun.logging.LogFactory;
+import org.radargun.traits.BasicOperations;
+
+/**
+ * Implementation of {@link BasicOperations} through the HTTP protocol, using the JAX-RS 2.0 client
+ * api with RESTEasy implementation.
+ *
+ * @author Alan Field &lt;afield@redhat.com&gt;
+ */
+public class RESTEasyHTTPOperations implements BasicOperations {
+   private static final Log log = LogFactory.getLog(RESTEasyHTTPOperations.class);
+   private final RESTEasyHTTPService service;
+
+   public RESTEasyHTTPOperations(RESTEasyHTTPService service) {
+      this.service = service;
+   }
+
+   @Override
+   public <K, V> HTTPCache<K, V> getCache(String cacheName) {
+      if (service.isRunning()) {
+         if (cacheName != null && (service.cacheName == null || !service.cacheName.equals(cacheName))) {
+            throw new UnsupportedOperationException();
+         }
+         return new HTTPCache<K, V>();
+      }
+      return null;
+   }
+
+   protected class HTTPCache<K, V> implements BasicOperations.Cache<K, V> {
+
+      @Override
+      public V get(K key) {
+         return getInternal(key).value;
+      }
+
+      private WrappedValue<V> getInternal(K key) {
+         V value = null;
+         EntityTag eTag = null;
+         if (service.isRunning()) {
+            String target = buildUrl(service.cacheName, key);
+            try {
+               Invocation get = service.getHttpClient().target(target).request().accept(service.getContentType())
+                     .buildGet();
+               Response response = get.invoke();
+               eTag = response.getEntityTag();
+               value = decodeByteArray(response.readEntity(byte[].class));
```

Comment:
Minor: you could decode this lazily, only if the subsequent operation (in getAndPut/getAndRemove) succeeds.